### ISSUE TITLE
Add a callbacks like 'CanChange'

### ISSUE THREAD

**Issue by daybson** [jpnurmi/flutter_spinbox#25]
I need to validate using some rules if the new value can be accepted before changed.

Would be nice to have callbacks like: `beforeIncrement. beforeDecrement, canChangeValue...`

Thank you!

**Comment by jpnurmi** [comment]
Hi, thanks for the suggestion. Could you give a sample use case to work with? So that I can understand better what kind of validation is needed.

**Comment by daybson** [comment]
Of course!
Well, I have an RPG app, and the user is creating a character. He has some Points to spend on attributes, like Strength, Speed, etc.
So:   Points = 15

And everytime he click to increase/decrease some attribute, first I need to look if he has at least 1 point to spend, and if he didn't overflow the attribute limit (maximum is 10), only after checks this the Strength can increase to +1. The min/max scenario is easy to setup, but the 'point buy' rules are more tricky wihtout a callback before the change to controll the 'allow/deny' value change.
And sometimes there are more rules involved... 

Something like this!

**Comment by jpnurmi** [comment]
Fixed by #27 - thanks!
